src/main/java/eu/deltasource/internship/abankingsystem/model/BankAccount.java

```diff
@@ -0,0 +1,119 @@
+package eu.deltasource.internship.abankingsystem.model;
+
+import eu.deltasource.internship.abankingsystem.enums.Currency;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public abstract class BankAccount {
+
+    private final Owner owner;
+
+    private final String iban;
+
+    private final Currency currency;
+
+    private Double amountAvailable;
+
+    private final char accountKey;
+
+    private List<Transaction> transferStatement;
+
+    private static final List<BankAccount> accounts = new ArrayList<>();
+
+    public BankAccount(Owner owner, String iban, Currency currency, double amountAvailable, char accountKey) {
+        this.owner = owner;
+        this.iban = normalizeIban(iban);
+        this.currency = currency;
+        this.amountAvailable = amountAvailable;
+        this.accountKey = normalizeAccountKey(accountKey);
+        this.transferStatement = new LinkedList<>();
+        accounts.add(this);
+    }
+
+    public static void countOfAccountOwnerHas(Owner owner) {
+        List<BankAccount> newListOwner = new ArrayList<>();
+        for (var singleAccount : accounts) {
+            if (singleAccount.getOwner().getName().equals(owner.getName())) {
+                newListOwner.add(singleAccount);
+            }
+        }
+        System.out.println(newListOwner);
+    }
+
+    public Owner getOwner() {
+        return owner;
+    }
+
+    public String getIban() {
+        return iban;
+    }
+
+    public Currency getCurrency() {
+        return currency;
+    }
+
+    public Double getAmountAvailable() {
+        return amountAvailable;
+    }
+
+    protected abstract String getAccountType();
+
+    public char getAccountKey() {
+        return accountKey;
+    }
+
+    private char normalizeAccountKey(final char accountKey) {
```

Comment:
I am using final parameters in a method to prevent the value of the parameter to be changed inside of the body of the method. This is more like a developer's protection not to make mistakes in the code. Not sure if it is a good convention.